src/jit/codegencommon.cpp

```diff
@@ -729,12 +729,7 @@ void Compiler::compUpdateLifeVar(GenTreePtr tree, VARSET_TP* pLastUseVars)
     // verifying that we see the var nodes in execution order
     if (ForCodeGen)
     {
-        if (tree->gtOper == GT_OBJ)
-        {
-            // The tree must have the particular form OBJ(ADDR(LCL)); no need to do the check below.
-            assert(indirAddrLocal != NULL);
-        }
-        else if (tree->OperIsIndir())
+        if (tree->OperIsIndir())
         {
```

Comment:
The comment on the #if above is confusing, especially the "with with" part. maybe you can fix it since you're here.